src/igvfd/tests/test_audit_measurement_set.py

```diff
@@ -1049,3 +1049,116 @@ def test_audit_missing_construct_library_set(
+    assert all(
+        error['category'] != 'missing barcode onlist information'
+        for error in res.json['audit'].get('WARNING', [])
+    )
+    # Check if a non-scRNAseq MeaSet without onlist files and method will trigger warnings.
+    res = testapp.get(measurement_set['@id'])
+    assert all(
+        error['category'] != 'missing barcode onlist information'
+        for error in res.json['audit'].get('WARNING', [])
+    )
+    # Patch the basic MeaSet to be scRNAseq without an onlist method or files to trigger audit
+    testapp.patch_json(
+        measurement_set['@id'],
+        {
+            'assay_term': assay_term_scrnaseq['@id']
+        }
+    )
+    res = testapp.get(measurement_set['@id'])
+    assert any(
+        error['category'] == 'missing barcode onlist information'
+        for error in res.json['audit'].get('WARNING', [])
+    )
+    # Patch the basic MeaSet with onlist method without onlist files to trigger missing onlist file audit
+    testapp.patch_json(
+        measurement_set['@id'],
+        {
+            'onlist_method': 'product'
+        }
+    )
+    res = testapp.get(measurement_set['@id'])
+    assert any(
+        error['category'] == 'missing barcode onlist information'
+        for error in res.json['audit'].get('WARNING', [])
+    )
+    # Patch the multiome MeaSet to be scRNAseq with onlist files without an onlist method to trigger missing onlist method audit
+    testapp.patch_json(
+        measurement_set_multiome['@id'],
+        {
+            'onlist_files': [tabular_file_onlist_1, tabular_file_onlist_2],
+            'assay_term': assay_term_scrnaseq['@id'],
+            'preferred_assay_term': 'scRNA-seq'
+        }
+    )
+    res = testapp.get(measurement_set_multiome['@id'])
+    assert any(
+        error['category'] == 'missing barcode onlist information'
+        for error in res.json['audit'].get('WARNING', [])
+    )
+
+
+def test_audit_inconsistent_barcode_onlist_information(testapp, measurement_set_one_onlist, measurement_set_two_onlist, tabular_file_onlist_1, tabular_file_onlist_2):
+    # Check if the measurement set fixture with one file and no combination method is audit-free
+    res = testapp.get(measurement_set_one_onlist['@id'])
+    assert all(
+        error['category'] != 'inconsistent barcode onlist information'
+        for error in res.json['audit'].get('WARNING', [])
+    )
+    # Check if the measurement set fixture with two file and combination method is audit-free
+    res = testapp.get(measurement_set_two_onlist['@id'])
+    assert all(
+        error['category'] != 'inconsistent barcode onlist information'
+        for error in res.json['audit'].get('WARNING', [])
+    )
+    # Add another onlist file to a MeaSet that is no-combination for onlist method.
+    testapp.patch_json(
+        measurement_set_one_onlist['@id'],
+        {
+            'onlist_files': [tabular_file_onlist_1, tabular_file_onlist_2]
```

Comment:
should be 'onlist_files': [tabular_file_onlist_1['@id'], tabular_file_onlist_2['@id']],